src/graphql/schema/buildObjectType.ts

```diff
@@ -391,16 +396,18 @@ function buildObjectType({
           if (id) {
             id = id.toString();
 
-            const relatedDocument = await context.req.payloadDataLoader.load(JSON.stringify([
-              relatedCollectionSlug,
-              id,
-              0,
-              0,
-              locale,
-              fallbackLocale,
-              false,
-              false,
-            ]));
+            const relatedDocument = await (context.req as PayloadRequest).payloadDataLoader.load(
+              {
+                collection: relatedCollectionSlug as string,
```

Comment:
type is string | string[]

Can this be fixed?

Reply:
Fixed in https://github.com/payloadcms/payload/tree/fix/dataloader-deadlock